### PR TITLE
[frontend] Create a separate upstream GitHub CI workflow for UI

### PR DESCRIPTION
**Pull request by nidhibhatg**
## What changes were proposed in this pull request?

- Create a separate github workflow with the following:
        Check out code
        Run commit title format check
        Prepare NPM (node/npm versions, caching etc)
        Run JS package.json version checker
        Run JS license checker
        Run NPM Absolute path detection
        Run JS lints
        Run Style lints
        Run JS tests
- When a cache is created by a workflow run triggered on a pull request, the cache is created for the merge ref (refs/pull/.../merge). Because of this, the cache will have a limited scope and can only be restored by re-runs of the pull request. It cannot be restored by the base branch or other pull requests targeting that base branch which could lead to **cache thrashing** when there are multiple PRs open. Hence added the clean up cache workflow that would be triggered when the PR is closed.

## How was this patch tested?

- Tested locally using https://github.com/nektos/act

### CODE REVIEW

File: .github/workflows/commitflow-ui.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Caching npm with setup node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
```
Comment (Harshg999) [inline .github/workflows/commitflow-ui.yml:25]: Question: Lowest we support is 16? Should we also test for other node versions with the matrix?

Curious: Will it have any significant effect though for other node versions if we don't test with them?

File: .github/workflows/commitflow-ui.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        cache: 'npm'
+
+    - name: Install npm
```
Comment (Harshg999) [inline .github/workflows/commitflow-ui.yml:28]: Nit: Maybe `Install npm dependencies`?

File: .github/workflows/commitflow-ui.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      run: npm ci
+
+    - name: run commit title format check
+      run: |
+        ./tools/ci/check_for_commit_message.sh 
+
+    - name : run npm version checker
```
Comment (Harshg999) [inline .github/workflows/commitflow-ui.yml:35]: Nit: bad indentation

File: .github/workflows/commitflow-ui.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - name: run js lint
+      run: |
+        npm run lint
+
+    - name: run style lint
+      run: |
+        npm run style-lint
+
+    - name: run js tests
+      run : |
```
Comment (Harshg999) [inline .github/workflows/commitflow-ui.yml:56]: Nit: bad indentation

File: desktop/core/src/desktop/js/reactComponents/createRootElements.js [hunk 5]
```diff
@@ -31,7 +31,7 @@ async function render(name, props, root) {
 }
 
 export async function createReactComponents(selector) {
-  // Find all DOM containers
+  // Find all DOM containers .
```
Comment (Harshg999) [inline desktop/core/src/desktop/js/reactComponents/createRootElements.js:34]: Nit: not required for the PR?

**Review by Harshg999 — COMMENTED** [review]
Nice work! I've added few review comments but overall looks good to me!

We are also seeing some `Unchanged files with check annotations`. Are these related?

File: .github/workflows/commitflow-ui.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+    paths:
+      - '**.js'
```
[the remark this answers is not in the document]
Comment (Harshg999) [reply .github/workflows/commitflow-ui.yml:11]: If its way too many files extensions to list, then does it makes sense to just remove it and run for every PR change? These checks might not be taking that much time anyway.

File: .github/workflows/commitflow-ui.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (nidhibhatg) [reply .github/workflows/commitflow-ui.yml:18]: I will make that change!

File: desktop/core/src/desktop/js/reactComponents/createRootElements.js (hunk 5 above)
Comment (nidhibhatg) [reply desktop/core/src/desktop/js/reactComponents/createRootElements.js]: No not required. It was purely for testing purposes

**Comment by nidhibhatg** [comment]
Closing the PR temporarily to test if the cache cleanup is being performed.

File: .github/workflows/commitflow-ui.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (nidhibhatg) [reply .github/workflows/commitflow-ui.yml:11]: .js, .jsx, .ts, .tsx, .mako, .less, .vue, .css, .less, .scss 
@JohanAhlen and @bjornalm am I missing any other extension with respect to UI? Could you please help me out with the same?

[the remark this answers is not in the document]
Comment (JohanAhlen) [reply .github/workflows/commitflow-ui.yml:11]: You can skip .mako and .css and you have .less twice. Also include package.json and package-lock.json

**Review by JohanAhlen — APPROVED** [review]
Nice one! Some minor comments but looks good!

Could you also remove the JS/Style checks from the other workflows?

File: .github/workflows/commitflow-ui.yml (hunk 1 above)
Comment (nidhibhatg) [reply .github/workflows/commitflow-ui.yml:26]: style listing with version 10 was failing, so currently will be using node-version: 14.

**Review by JohanAhlen — APPROVED** [review]
Nice and clean!

**Review by bjornalm — APPROVED** [review]
Nice work!

**Comment by Harshg999** [comment]
We need to see if we require cache cleaning if PR is closed or branch is deleted or not.

**Pull request merged**